lib/rules/no-restricted-exports.js

```diff
@@ -109,7 +112,15 @@ module.exports = {
         function checkExportedName(node) {
             const name = astUtils.getModuleExportName(node);
 
-            if (restrictedNames.has(name)) {
+            let matchesRestrictedNamePattern = false;
+
+            if (restrictedNamePattern) {
```

Comment:
```suggestion
            if (restrictedNamePattern && name !== "default") {
```

I think this is the only code change needed, but we should also add a few tests.